docs/operator-manual/applicationset/Generators-Plugin.md

```diff
@@ -0,0 +1,237 @@
+# Plugin Generator
+
+Plugins allow you to provide your own generator.
+
+* You can write in any language
+* Simple: a plugin just responds to RPC HTTP requests.
+* You can use it in a sidecar, or standalone deployment.
+* You can get your plugin running today, no need to wait 3-5 months for review, approval, merge and an Argo software
+  release.
+* You can combine it with Matrix or Merge.
+
+## Simple example
+
+Using a generator plugin without combining it with Matrix or Merge.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: myplugin
+spec:
+  generators:
+    - plugin:
+        # The name of the plugin used
+        name: my-plugin
+        # Specify the configMap where the plugin configuration is located
+        configMapRef: my-plugin
+        # You can pass parameters included in the RPC call
+        params:
```

Comment:
Maybe spell out "parameters"? We use the whole word in the `spec.source.helm.parameters` field in the Application CR. Might be good to model that.

Reply:
Yes why not, we can harmonize.